### PR TITLE
[tfhe] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
`tfhe` is a fast fully homomorphic encryption library over the Torus.
It currently only supports on Linux platform.

Relates issue #8352.

No feature need to test.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failure on osx due to this:
```
/Users/vagrant/azure_agent/_work/2/s/buildtrees/tfhe/src/v1.0.1-0e6bf01c41/src/libtfhe/fft_processors/nayuki/fft_processor_nayuki.cpp:15:35: error: implicit conversion from '_Complex double' to 'double' is not permitted in C++
        omegaxminus1[x]=cos(x*M_PI/N)-1. + sin(x*M_PI/N) * 1i; 
                       ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by NancyLi1013** [comment]
Since the failures on `osx`, I added the related message to remind users that this port currently doesn't support osx. 
So it can only be built successfully on `Linux `platform.

**Review by LilyWangL — APPROVED** [review]

File: ports/tfhe/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+vcpkg_fail_port_install(MESSAGE "${PORT} currently only supports on Linux platform" ON_TARGET "Windows" "osx")
```
Comment (dan-shaw) [inline ports/tfhe/portfile.cmake]: `${PORT} currently only supports Linux` might be clearer.

**Review by dan-shaw — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Please modify ci.baseline.txt to account for the windows failures.

File: ports/tfhe/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/tfhe/portfile.cmake]: Thanks for your review. Have updated.

**Pull request merged**